src/diffusers/loaders.py

```diff
@@ -1464,7 +1521,41 @@ def _maybe_map_sgm_blocks_to_diffusers(cls, state_dict, unet_config, delimiter="
         return new_state_dict
 
     @classmethod
-    def load_lora_into_unet(cls, state_dict, network_alphas, unet, low_cpu_mem_usage=None, _pipeline=None):
+    def _optionally_disable_offloading(cls, _pipeline):
+        """
+        Optionally removes offloading in case the pipeline has been already sequentially offloaded to CPU.
+
+        Args:
+            _pipeline (`DiffusionPipeline`):
+                The pipeline to disable offloading for.
+
+        Returns:
+            tuple:
+                A tuple indicating if `is_model_cpu_offload` or `is_sequential_cpu_offload` is True.
+        """
+        is_model_cpu_offload = False
+        is_sequential_cpu_offload = False
+
+        if _pipeline is not None:
+            for _, component in _pipeline.components.items():
+                if isinstance(component, nn.Module):
+                    if hasattr(component, "_hf_hook"):
+                        if not is_model_cpu_offload:
+                            is_model_cpu_offload = isinstance(component._hf_hook, CpuOffload)
+                        if not is_sequential_cpu_offload:
+                            is_sequential_cpu_offload = isinstance(component._hf_hook, AlignDevicesHook)
```

Comment:
This would have the same effect in two lines, but your version is probably easier to understand, just wanted to throw this out there.

```suggestion
                        is_model_cpu_offload |= isinstance(component._hf_hook, CpuOffload)
                        is_sequential_cpu_offload |= isinstance(component._hf_hook, AlignDevicesHook)
```

Reply:
Would prefer the current implementation as is in favor of easier readability.